src/ui/game.rs

```diff
@@ -1,93 +1,554 @@
 use std::time::Duration;
 
 use bevy::prelude::*;
-
-use crate::{
-    misc::HealthComponent,
+use thetawave_interface::{
+    health::HealthComponent,
+    objective::{NewObjectiveEvent, Objective},
     player::{PlayerComponent, PlayersResource},
-    run::{CurrentRunProgressResource, Objective},
     states::GameCleanup,
 };
 
+use crate::run::CurrentRunProgressResource;
+
 use super::BouncingPromptComponent;
 
-/// Tag for player health ui
+/// Tag for level ui
+#[derive(Component)]
+pub struct ObjectiveUI;
+
+/// Tag for level ui
+#[derive(Component)]
+pub struct ObjectiveLabelUI;
+
+/// Tag for level ui
+#[derive(Component)]
+pub struct PowerGlowUI(Timer);
+
+#[derive(Component)]
+pub struct StatBarLabel;
+
+#[derive(Component)]
+pub struct AbilityChargingUI;
```

Comment:
Dont make things public unless you expect other plugins/systems to use it. These can be private to the module. Really the only things that should need to be public are systems, and a couple other bevy things.